server/models/Document.js

```diff
@@ -32,12 +32,17 @@ const createRevision = (doc, options = {}) => {
   // we don't create revisions if identical to previous
   if (doc.text === doc.previous('text')) return;
 
-  return Revision.create({
-    title: doc.title,
-    text: doc.text,
-    userId: doc.lastModifiedById,
-    documentId: doc.id,
-  });
+  return Revision.create(
+    {
+      title: doc.title,
+      text: doc.text,
+      userId: doc.lastModifiedById,
+      documentId: doc.id,
+    },
+    {
+      transaction: options.transaction,
```

Comment:
Requirement, otherwise we saw a race condition with creating revisions and attempting to read them in the backlinks service.